index.bs

```diff
@@ -257,28 +259,72 @@ spec:css-syntax-3;
   This document depends on the Infra Standard for a number of foundational concepts used in its
   algorithms and prose [[!INFRA]].
 
+
+  ### Infrastructure Algorithms ### {#sctn-infra-algorithms}
+
+  <!-- NOTE: '<h4>' maps to the same heading level as three hashs '###', thus using '<h5>' for the next section -->
+  <h5 id="algorithm-same-origin-with-ancestors" algorithm>Same-Origin with its Ancestors</h5>
+
   An [=environment settings object=] (|settings|) is <dfn noexport>same-origin with its
   ancestors</dfn> if the following algorithm returns `true`:
 
-  1.  If |settings| has no [=environment settings object/responsible document=],
-      return `false`.
+  <ol class="algorithm">
+    1.  If |settings| has no [=environment settings object/responsible document=],
+        return `false`.
+
+    2.  Let |document| be |settings|' [=environment settings object/responsible document=].
+
+    3.  If |document| has no [=document/browsing context=], return `false`.
 
-  2.  Let |document| be |settings|' [=environment settings object/responsible document=].
+    4.  Let |origin| be |settings|' [=environment settings object/origin=].
 
-  3.  If |document| has no [=document/browsing context=], return `false`.
+    5.  Let |current| be |document|'s [=document/browsing context=].
 
-  4.  Let |origin| be |settings|' [=environment settings object/origin=].
+    6.  While |current| has a [=parent browsing context=]:
 
-  5.  Let |current| be |document|'s [=document/browsing context=].
+        1.  Set |current| to |current|'s [=parent browsing context=].
 
-  6.  While |current| has a [=parent browsing context=]:
+        2.  If |current|'s [=active document=]'s [=origin=] is not [=same origin=] with |origin|,
+            return `false`.
 
-      1.  Set |current| to |current|'s [=parent browsing context=].
+    7.  Return `true`.
+  </ol>
+
+  ### Registry: Credential Type to Options Member Identifier Mappings ### {#sctn-registry-cred-type-to-options-member-identifier}
+
+  This registry maps [=credential types=] (i.e., {{Credential/[[type]]}} values) to the
+  [=dictionary member=] [=identifiers=] used in
+  {{CredentialCreationOptions}} and {{CredentialRequestOptions}} (i.e., "Options")
+  by their specifications.
+
+  Note: This registry is used by the [=relevant credential interface objects=] algorithm.
+
+
+  <table class="complex data longlastcol">
+    <tr>
+        <th>Credential Type<br>
+        (in alphabetical order)</th>
+        <th>Options Member Identifier</th>
+        <th>Specification</th>
+    </tr>
+    <tr>
+        <td>federated</td>
+        <td>federated</td>
+        <td>This specification: [[#federated]]</td>
+    </tr>
+    <tr>
+        <td>password</td>
+        <td>password</td>
+        <td>This specification: [[#passwords]]</td>
+    </tr>
+    <tr>
+        <td>public-key</td>
+        <td>publicKey</td>
+        <td>[[WEBAUTHN]]</td>
```

Comment:
Add OTP: https://wicg.github.io/web-otp/#dom-otpcredential-type-slot

Reply:
another good catch, thx, done in b2ba40f